nx_parallel/algorithms/approximation/connectivity.py

```diff
@@ -0,0 +1,51 @@
+"""Parallel implementations of fast approximation for node connectivity"""
+import itertools
+from joblib import Parallel, delayed
+import nx_parallel as nxp
+from networkx.algorithms.approximation.connectivity import local_node_connectivity
+
+__all__ = [
+    "all_pairs_node_connectivity",
+]
+
+
+def all_pairs_node_connectivity(G, nbunch=None, cutoff=None):
+    def _calculate_all_pairs_node_connectivity_subset(chunk):
+        for u, v in chunk:
+            k = local_node_connectivity(G, u, v, cutoff=cutoff)
+            all_pairs[u][v] = k
+            if not directed:
+                all_pairs[v][u] = k
```

Comment:
> It might be worth defining this function based on the value of directed

here, I think you mean having 2 helper functions(one for when `directed` is True and one for when it's False), but i think passing `directed` as an argument in one function might be better.

Either way, I think the impact on the performance would be negligible but passing `directed` as an argument in the inside function might decrease the number of lookups in the outer namespace, so I think it would be better than having 2 different functions as that would reduce the readability of the code. Let me know what you think.